### ISSUE TITLE
Enable the use of Elastic Shield

### ISSUE THREAD

**Issue by devth** [clojurewerkz/elastisch#226]
https://www.elastic.co/guide/en/shield/current/_using_elasticsearch_java_clients_with_shield.html

**Comment by timgluz** [comment]
i'll digg into that;

**Comment by timgluz** [comment]
Hi,

I'm almost done with this task, but I have a few question before i start preparing PR.
Is it fine to add the additional dependency for the `Shield` plugin? 
If not, should I publish it as a separate package?

**Comment by michaelklishin** [comment]
@timgluz most likely it is, can you elaborate? Is it an open source client library from Elastic? under what license?

**Comment by michaelklishin** [comment]
Looks like it has a library part but is not open source. I cannot find any license or usage information.
If so, it should be an optional dependency (e.g. something that applications list as dependencies if they want to use Shield). We can also consider making it a separate ClojureWerkz project.

**Comment by timgluz** [comment]
Yes, it's a proprietary extension, managed and owned by Elastic and first install includes only 30days trial; [source](https://www.elastic.co/guide/en/shield/current/license-management.html)

That's why I am thinking of creating a new project `clojurewerkz.elastisch.shield`, which could be added as the additional dependency if people want to work with the `shield`. And another reason is testing - it requires own separate Docker image with custom configuration for users and roles;

**Comment by michaelklishin** [comment]
@timgluz alright. Should I set up a new ClojureWerkz repo for you then?

**Comment by timgluz** [comment]
Yes, please;

**Comment by michaelklishin** [comment]
@timgluz done: https://github.com/clojurewerkz/elastisch.shield. I'm making sure you have push permissions now.

One question I have w.r.t. Shield support is how we are going to do integration testing if it's a commercial plugin. Not having tests really sucks.

**Comment by timgluz** [comment]
Thanks!

I'm adding tests for all the new functionalities and i'm also going to re-use tests from `elastisch` to test that the Shield clients work as `rest-client` and `native-client`.

I'm also going to set up travis for the new project;
